README.md

```diff
@@ -420,152 +734,188 @@ A candidate will be selected if:
-                     ranking_metric="cosine", 
-                     selection_threshold=0.51, 
+    candidate_ranker(candidate_scenario="./combined/candidates_test",
+                     query=["DeezyMatch", "kasra", "fede", "mariona"],
+                     ranking_metric="faiss", 
+                     selection_threshold=5., 
                      num_candidates=1, 
-                     search_size=4, 
-                     output_filename="test_candidates_deezymatch", 
+                     search_size=100, 
+                     output_path="ranker_results/test_candidates_deezymatch_on_the_fly", 
                      pretrained_model_path="./models/finetuned_test001/finetuned_test001.model", 
                      pretrained_vocab_path="./models/finetuned_test001/finetuned_test001.vocab", 
                      number_test_rows=20)
 ```
 
-Or via command line:
+The candidate ranker can be initialised, to be used multiple times, by running:
 
-```bash
-DeezyMatch --deezy_mode candidate_ranker -comb ./combined/test -rm cosine -t 0.51 -n 1 -sz 4 -o test_candidates_deezymatch -mp ./models/finetuned_test001/finetuned_test001.model -v ./models/finetuned_test001/finetuned_test001.vocab -tn 20
-```
+```python
+from DeezyMatch import candidate_ranker_init
 
-## Installation
+# initializing candidate_ranker via candidate_ranker_init
+myranker = candidate_ranker_init(candidate_scenario="./combined/candidates_test",
+                                 query=["DeezyMatch", "kasra", "fede", "mariona"],
+                                 ranking_metric="faiss", 
+                                 selection_threshold=5., 
+                                 num_candidates=1, 
+                                 search_size=100, 
+                                 output_path="ranker_results/test_candidates_deezymatch_on_the_fly", 
+                                 pretrained_model_path="./models/finetuned_test001/finetuned_test001.model", 
+                                 pretrained_vocab_path="./models/finetuned_test001/finetuned_test001.vocab", 
+                                 number_test_rows=20)
+```
 
-We strongly recommend installation via Anaconda:
+The content of `myranker` can be printed by:
 
-1. Refer to [Anaconda website and follow the instructions](https://docs.anaconda.com/anaconda/install/).
+```python
+print(myranker)
+```
 
-2. Create a new environment for DeezyMatch
+which results in:
 
 ```bash
-conda create -n py37deezy python=3.7
-```
+-------------------------
+* Candidate ranker params
+-------------------------
 
-3. Activate the environment:
+Queries are based on the following list:
+["DeezyMatch", "kasra", "fede", "mariona"]
 
-```bash
-conda activate py37deezy
+candidate_scenario:     ./combined/candidates_test
+---Searching params---
+num_candidates:         1
+ranking_metric:         faiss
+selection_threshold:    5.0
+search_size:            100
+number_test_rows:       20
+---I/O---
+input_file_path:        default
```

Comment:
`default` here means "the input file that is inside the `candidate_scenario` folder"? If so, could we print something like this?
```
input_file_path: default (path: ./combined/candidates_test/input_dfm.yaml)
```

Reply:
Done

Reply:
Sorry, the answer to your question is yes, and implemented accordingly.

Reply:
Great, thank you!